actions/run-nextstrain-build-on-aws-batch/action.yaml

```diff
@@ -0,0 +1,68 @@
+name: Run Nextstrain build on AWS Batch
+description: >-
+  This GitHub Actions action is intended to be called by workflows in our other
+  repos when they need to run a nextstrain build on AWS Batch.
+
+inputs:
+  runtime-envvar-names:
+    description: >-
+      A list of environment variable names for environment variables that
+      should be passed to the build. The envvar names should be separated by
+      a space, e.g. "GITHUB_RUN_ID SLACK_TOKEN SLACK_CHANNELS"
+    type: string
+    required: false
+
+  cpus:
+    description: >-
+      Number of CPUs/cores/threads/jobs to utilize at once.
+      Will be passed to the `--cpus` option for `nextstrain build` and the
+      `--cores` option for Snakemake.
+    type: number
+    default: 4
+    required: false
+
+  memory:
+    description: >-
+      Amount of memory to make available to the build. Passed to the
+      `--memory` option for `nextstrain build`.
+    type: string
+    required: false
+
+  build-args:
+    description: >-
+      Additional command-line arguments to pass to `nextstrain build` after
```

Comment:
Is choosing these arguments identical to what you would provide to `nextstrain build` if it were calling snakemake for you (rather than us using `--exec` and then calling it ourselves? This came up recently so I think some docs here would be really helpful.

Reply:
Ah, thanks for asking. I'll update the description to be clearer that these are additional arguments passed directly to Snakemake.

Reply:
> I'll update the description to be clearer that these are additional arguments passed directly to Snakemake.

In a similar vein to my above comments, I would target this description to explaining how to go from running  `snakemake ...` to knowing what parts of the `...` I should include here. For instance, I should remove `--cores N`, and probably some other things right?

Perhaps it should also explain how to modify the arguments from `nextstrain build ...` to use here, as that's where uses may be coming from. In that case, however, isn't the runtime they are using important to know as it would affect the changes you need to make?

(I think no problem having verbose descriptions, in fact I prefer them!)

Reply:
> Perhaps it should also explain how to modify the arguments from nextstrain build ... to use here, as that's where uses may be coming from.

Ah, this input is only for any additional Snakemake arguments, it does not pass arguments to Nextstrain CLI. I'll change the wording of the description to be less confusing.

Reply:
Yup - apologies - I could have been more precise. If I have a command running my workflow such as (any of):

```sh
snakemake --cores 1 -pf some/taget.x
nextstrain build someDirectory --resources A=B --profile C/D # what's my runtime? Does it matter?
```

I think the information required to map these to the `build-args` of this action should be provided by this action. However if the action is to be "basically a wrapper around the Nextstrain CLI" then "Additional command-line arguments to pass to `nextstrain build`" is probably enough, although I'd use the same language as `nexstrain` does: "Additional arguments to pass to the executed program".

Reply:
@jameshadfield [wrote](https://github.com/nextstrain/.github/pull/39#discussion_r1210871142):

> […] then "Additional command-line arguments to pass to `nextstrain build`" is probably enough, although I'd use the same language as `nexstrain` does: "Additional arguments to pass to the executed program".

Hmm. I don't think using the exact same language makes sense out of context. "Additional arguments to pass to the executed program" makes sense in the context of the rest of the `nextstrain build --help[-all]` output¹, which also says things like "… will be run with snakemake" and the "Program to execute inside the runtime (default: snakemake)". I think different language is warranted in the different context here.

¹ All the context used to be part of just `--help`, but that changed when `--help-all` was added and the description of the additional args should probably change too.